core/src/main/java/org/apache/accumulo/core/client/summary/summarizers/EntryLengthSummarizer.java

```diff
@@ -0,0 +1,135 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License");you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.client.summary.summarizers;
+
+import java.math.RoundingMode;
+import java.util.Map;
+
+import org.apache.accumulo.core.client.summary.Summarizer;
+import org.apache.accumulo.core.client.summary.SummarizerConfiguration;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+
+import com.google.common.math.IntMath;
+
+/**
+ * Summarizer that computes summary information about field lengths.
+ * Specifically key length, row length, family length, qualifier length, visibility length, and value length.
+ * Incrementally computes minimum, maximum, count, sum, and log2 histogram of the lengths.
+ */
+public class EntryLengthSummarizer implements Summarizer {
+
+  /* Helper function that calculates the various statistics that is used for the Collector methods.*/
+  private static class LengthStats {
+    private long min = Long.MAX_VALUE;
+    private long max = Long.MIN_VALUE;
+    private long sum = 0;
+    private long[] counts = new long[32];
+
+    private void accept(int length) {
+      int idx;
+
+      if (length < min) {
+        min = length;
+      }
+
+      if (length > max) {
+        max = length;
+      }
+
+      sum += length;
+
+      if (length == 0) {
+        idx = 0;
+      } else {
+        idx = IntMath.log2(length, RoundingMode.HALF_UP);
+      }
+
+      counts[idx]++;
+    }
+
+    void summarize (String prefix, StatisticConsumer sc) {
+      sc.accept(prefix+".min", (min != Long.MAX_VALUE ? min:0));
+      sc.accept(prefix+".max", (max != Long.MIN_VALUE ? max:0));
+      sc.accept(prefix+".sum", sum);
+
+      for (int i = 0; i < counts.length; i++) {
+        if (counts[i] > 0) {
+          sc.accept(prefix+".logHist."+i, counts[i]);
+        }
+      }
+    }
+
+  }
+
+  /* Helper function for merging that is used by the Combiner. */
+  private static void merge(String prefix, Map<String, Long> stats1, Map<String,Long> stats2 ) {
+    stats1.merge(prefix+".min", stats2.get(prefix+".min"), Long::max);
+    stats1.merge(prefix+".max", stats2.get(prefix+".max"), Long::max);
+    stats1.merge(prefix+".sum", stats2.get(prefix+".sum"), Long::sum);
+    stats2.forEach((k,v) -> stats1.merge(k, v, Long::sum));
```

Comment:
It seems like this will sum everything, even the min and max keys.  I suspect this is only intended to sum log histogram keys?    If so it needs a check for the key before doing merge.  Also, it seems like the test should catch this?

Reply:
I'm not sure I can articulate this correctly but I will try. I believe that since their is only one value passed into the merge functions that the remapping function recomputes the same value passed in. If that makes sense. Either way though, I played around with that during my initial coding, and it doesn't in this particular case seem to make any difference. I could make that parameter "null"  if that would be better, the output will be the same according to the test.

Reply:
Now looking at it again it does seem it shouldn't be working but why it is I don't exactly know, but again whether max, min, sum or null it returns sams values.

Reply:
I wasn't sure why the test were not catching this so I cloned the branch to experiment locally.  Realized the test are not calling this code, I completely missed that.  Below are the beginnings of a test I wrote while looking into this.

```java
  @Test
  public void testFoo() {
    SummarizerConfiguration sc = SummarizerConfiguration.builder(EntryLengthSummarizer.class).build();
    EntryLengthSummarizer entrySum = new EntryLengthSummarizer();

    Collector collector1 = entrySum.collector(sc);
    collector1.accept(new Key("1","f1","q1"), new Value("v1"));
    collector1.accept(new Key("1234","f1","q1"), new Value("v111"));
    collector1.accept(new Key("12345678","f1","q1"), new Value("v111111"));

    Map<String, Long> stats1 = new TreeMap<>();
    collector1.summarize(stats1::put);

    Collector collector2 = entrySum.collector(sc);
    collector2.accept(new Key("5432","f11","q12"), new Value("2"));
    collector2.accept(new Key("12","f11","q1234"), new Value("12"));
    collector2.accept(new Key("12","f11","q11234567"), new Value("4444"));

    Map<String, Long> stats2 = new TreeMap<>();
    collector2.summarize(stats2::put);

    System.out.println("========= Stats 1");
    stats1.entrySet().forEach(System.out::println);
    System.out.println("========= Stats 2");
    stats2.entrySet().forEach(System.out::println);

    Combiner combiner = entrySum.combiner(sc);
    combiner.merge(stats1, stats2);

    System.out.println("========= Merged");
    stats1.entrySet().forEach(System.out::println);
  }
```

Reply:
Oh wow that is complete oversight on my part, I was so focused on the Collector, I didn't test the Combiner. I will write up some test for that also and fix according. Thanks for pointing this out.

Reply:
Hmmm so for the Combiner I think I understand what should happen for the values that are singular; min,max,sum, and total, but for the log2Histo should it just add the value of the 2nd map to the 1st and combine the total for keys that match?